backend/controllers/authController.js

```diff
@@ -22,129 +20,106 @@ const verifyToken = async (req, res, next) => {
     });
   }
 
-  // Verify access token
-  jwt.verify(oldToken, process.env.JWT_SECRET, (err, decoded) => {
-    if (err) {
-      // If access token is invalid, verify the refresh token
-      jwt.verify(oldRefreshToken, process.env.JWT_SECRET, async (refreshErr, refreshDecoded) => {
-        if (refreshErr) {
-          return res.status(401).json({
-            status: "fail",
-            message: "Invalid refresh token",
-          });
-        }
-
-        // Generate new tokens
-        const newToken = generateToken(
-          { id: refreshDecoded.id },
-          process.env.JWT_SECRET,
-          process.env.JWT_EXPIRES_IN
-        );
-        const newRefreshToken = generateToken(
-          { id: refreshDecoded.id },
-          process.env.JWT_SECRET,
-          process.env.JWT_REFRESH_EXPIRES_IN
-        );
-
-        // Set new tokens in cookies
-        res.cookie("token", newToken, {
-          httpOnly: true,
-          secure: process.env.NODE_ENV === 'production',
-        });
-        res.cookie("refreshToken", newRefreshToken, {
-          httpOnly: true,
-          secure: process.env.NODE_ENV === 'production',
-        });
-
-        return res.status(200).json({
-          status: "success",
-          message: "Token refreshed successfully",
-        });
-      });
-    } else {
-      // Access token is valid
-      return res.status(200).json({
-        status: "success",
-        message: "Token is valid",
-      });
-    }
-  });
+  try {
+    // Decode refresh token
+    const refreshDecoded = jwt.decode(oldRefreshToken);
+
+    // Generate new tokens
+    const newToken = generateToken(
+      { id: refreshDecoded.id },
+      process.env.JWT_SECRET,
+      process.env.JWT_EXPIRES_IN
+    );
+    const newRefreshToken = generateToken(
+      { id: refreshDecoded.id },
+      process.env.JWT_SECRET,
+      process.env.JWT_REFRESH_EXPIRES_IN
+    );
+
+    // Set new tokens in cookies
+    res.cookie("token", newToken, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+    });
+    res.cookie("refreshToken", newRefreshToken, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+    });
+
+    return res.status(200).json({
+      status: "success",
+      message: "Token refreshed successfully",
+    });
+  } catch (error) {
+    return res.status(401).json({
+      status: "fail",
+      message: "Invalid refresh token",
+    });
+  }
 };
 
-// Signup controller
+// signup controller
 const signup = async (req, res, next) => {
-  const { username, email, password, confirmPassword, firstName, lastName, contact } = req.body;
+  const body = req.body;
+  console.log(body);
```

Comment:
```suggestion

```
best practice to remove console statements when pushing code

Reply:
good catch